web/src/integration_url_modal.js

```diff
@@ -57,25 +77,53 @@ export function show_generate_integration_url_modal(api_key) {
             update_url();
         });
 
-        function update_url() {
+        function update_url(update_events = false) {
             selected_integration = integration_input_dropdown_widget.value();
             if (selected_integration === default_integration_option.unique_id) {
                 $integration_url.text(default_url_message);
                 $dialog_submit_button.prop("disabled", true);
                 return;
             }
 
+            const params = new URLSearchParams({api_key});
+
+            const selected_integration_data = realm.realm_incoming_webhook_bots.find(
+                (bot) => bot.name === selected_integration,
+            );
+            const all_event_types = selected_integration_data.all_event_types;
+
+            if (
+                all_event_types !== null &&
+                !update_events &&
+                $show_integration_events.prop("checked")
+            ) {
+                all_event_types.sort((a, b) => a.localeCompare(b));
+                const events = render_integration_events({
+                    events: all_event_types,
+                });
+                $("#integration-events-container").empty().append(events);
+            } else if (!update_events && all_event_types === null) {
+                $("#integration-events-toggle-wrapper").addClass("hide");
+                $("#integration-events-ui").addClass("hide");
+                $("#integration-events-container").empty();
+                $(".integration-event-checkbox").prop("checked", false);
+                $show_integration_events.prop("checked", false);
+            }
+
+            if (all_event_types !== null) {
+                $("#integration-events-toggle-wrapper").removeClass("hide");
+            }
+
             const stream_id = stream_input_dropdown_widget.value();
             const topic_name = $topic_input.val();
-
-            const params = new URLSearchParams({api_key});
```

Comment:
There's no need to change where this is called in the function. The params constant isn't referenced until after this line.

Also, for readability, I'd put the events additions after the stream and topic constants. So the flow would be:
- Get the selected integration
- Check for the early return
- Get the stream and topic constants
- Get the events constant and check for showing/hiding elements
- Create the params constant
- Set the stream, topic and events parameters
- etc.

Reply:
This is still not quite the flow. You currently have ...

- Get the selected integration
- Check for the early return
- Get the stream and topic constants
- Create the params constant
- Get the events constant and check for showing/hiding elements
- Set the stream, topic and events parameters
- etc.